### PR TITLE
Statusline for neovim 0.10 

### PR DESCRIPTION
**Pull request by yoplat**
Modified statusline lsp progress and servers to work with neovim 0.10

### CODE REVIEW

File: lua/nvchad/statusline/minimal.lua [hunk 1]
```diff
@@ -109,35 +110,27 @@ end
...
 M.LSP_progress = function()
-  if not rawget(vim, "lsp") or vim.lsp.status then
-    return ""
-  end
-
-  local Lsp = vim.lsp.util.get_progress_messages()[1]
+  local status = utils.get_message()
 
-  if vim.o.columns < 120 or not Lsp then
+  if not rawget(vim, "lsp") or vim.o.columns < 120 or not status then
```
Comment (lucario387) [inline lua/nvchad/statusline/minimal.lua]: don't think rawget(vim, "lsp") is really needed nowadays

**Comment by siduck** [comment]
I'm not willing to support lsp progress for v0.10 until v0.10 becomes the latest stable release. For the time being, just override statusline module

**Comment by yoplat** [comment]
> I'm not willing to support lsp progress for v0.10 until v0.10 becomes the latest stable release. For the time being, just override statusline module

Yeah that’s what I did, I just thought it would be useful for anyone who uses nightly like me. Close the issue if you don’t need it

**Comment by yoplat** [comment]
I did a rework on how the lsp progress icon is displayed using the percentage instead of the current time, thought it was more intuitive

**Comment by siduck** [comment]
> > I'm not willing to support lsp progress for v0.10 until v0.10 becomes the latest stable release. For the time being, just override statusline module
> 
> Yeah that’s what I did, I just thought it would be useful for anyone who uses nightly like me. Close the issue if you don’t need it

no reason to close it. It'll just be open until v0.10 releases, I'll merge this

**Comment by yoplat** [comment]
> > > I'm not willing to support lsp progress for v0.10 until v0.10 becomes the latest stable release. For the time being, just override statusline module
> > 
> > 
> > Yeah that’s what I did, I just thought it would be useful for anyone who uses nightly like me. Close the issue if you don’t need it
> 
> no reason to close it. It'll just be open until v0.10 releases, I'll merge this

So is v3.0 scheduled to be on 0.9 or where you thinking of releasing it when 0.10 is out?

**Comment by siduck** [comment]
> > > > I'm not willing to support lsp progress for v0.10 until v0.10 becomes the latest stable release. For the time being, just override statusline module
> > > 
> > > 
> > > Yeah that’s what I did, I just thought it would be useful for anyone who uses nightly like me. Close the issue if you don’t need it
> > 
> > 
> > no reason to close it. It'll just be open until v0.10 releases, I'll merge this
> 
> So is v3.0 scheduled to be on 0.9 or where you thinking of releasing it when 0.10 is out?

New releases in nvchad have new features. So its not really tied to nvim versions. And this PR isnt a feature, it just updates the nvim api of doing things.

File: lua/nvchad/statusline/default.lua [hunk 2]
```diff
@@ -103,42 +104,30 @@ end
 
 -- LSP STUFF
 M.LSP_progress = function()
-  if not rawget(vim, "lsp") or vim.lsp.status then
-    return ""
-  end
+  local status = utils.get_message()
```
Comment (lucario387) [inline lua/nvchad/statusline/default.lua]: what is this get_message() or am I clueless

Comment (yoplat) [reply lua/nvchad/statusline/default.lua]: > what is this get_message() or am I clueless

Added it to utils.lua file, I don’t think there is a way to get the progress message like before so I created a similar function

Comment (lucario387) [reply lua/nvchad/statusline/default.lua]: So where's the get_message for NvChad then? Because rn I don't see any branch with that function

Comment (yoplat) [reply lua/nvchad/statusline/default.lua]: Screenshot 2023-10-10 at 16 26 18

Are you sure? This is what I see

Comment (lucario387) [reply lua/nvchad/statusline/default.lua]: All I see is that you are having `local utils = require ("core.utils")` and then using `utils.get_message`, which leads to the question of where that function actually lies

Comment (yoplat) [reply lua/nvchad/statusline/default.lua]: > All I see is that you are having `local utils = require ("core.utils")` and then using `utils.get_message`, which leads to the question of where that function actually lies

I seriously don't know what to tell you, I see the file in the "files changed" tab in github and in lua/nvchad/utils.lua there is the get_message function

Comment (yoplat) [reply lua/nvchad/statusline/default.lua]: I'm not that used to pull requests so I might have made something wrong, I merged the current v3.0 branch and the utils.lua file still contains the function

Comment (lucario387) [reply lua/nvchad/statusline/default.lua]: You are not requiring nvchad/utils.lua file from what I can see

```lua
local utils = require("core.utils") 
```
means utils is the returned module from `lua/core/utils.lua`, not `lua/nvchad/utils.lua`

Comment (yoplat) [reply lua/nvchad/statusline/default.lua]: > You are not requiring nvchad/utils.lua file from what I can see
> 
> ```lua
> local utils = require("core.utils") 
> ```
> 
> means utils is the returned module from `lua/core/utils.lua`, not `lua/nvchad/utils.lua`

Whooops! Now I get it. Fixing it right now thanks

File: lua/nvchad/statusline/default.lua [hunk 3]
```diff
@@ -118,7 +118,7 @@ end
 
 -- LSP STUFF
 M.LSP_progress = function()
-  local status = utils.get_message()
+  local status = require("utils").get_message()
```
Comment (lucario387) [inline lua/nvchad/statusline/default.lua]: Wrong
```suggestion
  local status = require("nvchad.utils").get_message()
```

**Review by lucario387 — CHANGES_REQUESTED** [review]

File: lua/nvchad/statusline/default.lua (hunk 3 above)
Comment (yoplat) [reply lua/nvchad/statusline/default.lua]: I'm stupid lol

**Review by lucario387 — APPROVED** [review]

**Pull request closed without merging**